src/ORM/Table.php

```diff
@@ -594,30 +592,6 @@ protected function checkAliasLengths(): void
         }
     }
 
-    /**
-     * Override this function in order to alter the schema used by this table.
-     * This function is only called after fetching the schema out of the database.
-     * If you wish to provide your own schema to this table without touching the
-     * database, you can override schema() or inject the definitions though that
-     * method.
-     *
-     * ### Example:
-     *
-     * ```
-     * protected function _initializeSchema(\Cake\Database\Schema\TableSchemaInterface $schema) {
-     *  $schema->setColumnType('preferences', 'json');
-     *  return $schema;
-     * }
-     * ```
-     *
-     * @param \Cake\Database\Schema\TableSchemaInterface $schema The table definition fetched from database.
-     * @return \Cake\Database\Schema\TableSchemaInterface the altered schema
-     */
-    protected function _initializeSchema(TableSchemaInterface $schema): TableSchemaInterface
```

Comment:
Couldn't they just override `getSchema()` then? Call `parent::getSchema()` and then use the instance as needed. It would achieve the same thing.

Reply:
> Couldn't they just override getSchema() then? Call parent::getSchema() and then use the instance as needed. It would achieve the same thing. The only benefit of this hook is one doesn't have to do the parent call.

Totally could do that. That gives a simple way for user-land code to upgrade in 4.x as well. I'll open a pull request for the deprecation.

Reply:
So you customize in the getter? How would they customize once?

Reply:
Ah this method is builder cum getter :(. So they would have to check `$this->_schema` too.

Reply:
I don't think checking the `_schema` is that important. Adding a column to a table will overwrite the existing column should it exist. Remapping a column is a cheap operation in comparison to the cost of schema reflection or reading from cache.

Reply:
> I don't think checking the _schema is that important.

Right, it isn't. The parent call will handle the null check for the property. I wasn't thinking straight earlier :)